python/ray/experimental/collective/allreduce.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from typing import List, Optional
+
+import ray
+from ray.dag.collective_node import CollectiveGroup, CollectiveOutputNode
+from ray.dag.constants import (
+    BIND_INDEX_KEY,
+    COLLECTIVE_GROUP_KEY,
+    PARENT_CLASS_NODE_KEY,
+)
+from ray.dag.dag_node import DAGNode
+from ray.experimental.channel.torch_tensor_type import TorchTensorType
+from ray.util.collective import types
+
+logger = logging.getLogger(__name__)
+
+
+class AllReduceWrapper:
+    """Wrapper for NCCL all-reduce."""
+
+    def bind(
+        self,
+        input_nodes: List["DAGNode"],
+        op: types.ReduceOp,
+        type_hint: Optional[TorchTensorType] = None,
```

Comment:
Instead of taking `type_hint`, it would be better to directly take the same keyword arguments, so that way:
- it's less code
- you can have different default values compared to TorchTensorType

Reply:
Done.